nion/swift/LinePlotCanvasItem.py

```diff
@@ -430,31 +430,46 @@ def __view_to_intervals(self, data_and_metadata: DataAndMetadata.DataAndMetadata
         else:
             left = right = 0.0
 
-        data_size = int(data_and_metadata.data_shape[-1])
-        left_channel = int(numpy.floor(left * data_size))
-        right_channel = int(numpy.ceil(right * data_size))
-        if left_channel > right_channel:
-            right_channel, left_channel = left_channel, right_channel
-        if left_channel not in range(data_size):
-            left_channel = 0
-        if right_channel not in range(data_size):
-            right_channel = data_size
-
-        channel_data = data_and_metadata.data
-        assert channel_data is not None
-        if channel_data[..., left_channel:right_channel].size > 0:
-            data_min = numpy.amin(channel_data[..., left_channel:right_channel])
-            data_max = numpy.amax(channel_data[..., left_channel:right_channel])
-        else:
-            data_min = numpy.amin(channel_data[..., :])
-            data_max = numpy.amax(channel_data[..., :])
+        top_level_data_size = int(data_and_metadata.data_shape[-1])
+        top_level_offset = data_and_metadata.intensity_calibration.offset
+        top_level_scale = data_and_metadata.intensity_calibration.scale
+
+        y_minimums = []
+        y_maximums = []
+
+        for display_value in self.__display_values_list:
+            display_value = typing.cast(DisplayItem.DisplayValues, display_value)
+            adjusted_data_and_metadata = display_value.adjusted_data_and_metadata
+            assert adjusted_data_and_metadata is not None
+            adjusted_data = adjusted_data_and_metadata.data
+            data_size = adjusted_data_and_metadata.data_shape[-1]
+            offset = adjusted_data_and_metadata.intensity_calibration.offset
+            scale = adjusted_data_and_metadata.intensity_calibration.scale  # Scale *should* handle frequencies
+            scaling_factor = scale / top_level_scale
+            offset_factor = int(numpy.round(top_level_offset - (offset * scaling_factor)))
+
+            left_channel = int(numpy.floor(top_level_data_size * left * scaling_factor) + offset_factor)
```

Comment:
Beware of mixing calibrations -- channels (x-values) are calibrated using `dimensional_calibrations` (0 index in most cases); intensity (y-values) are calibrated using `intensity_calibration`. `scaling_factor` is based on an intensity calibration (y-value), but used to calculate a channel (x-value).

Reply:
Also, you'll want to just use the `Calibration` object methods directly:
https://github.com/nion-software/niondata/blob/f17642bc0cdee2c5d8dc77144486171d005fba3c/nion/data/Calibration.py#L112-L140